src/gromacs/nblib/box.cpp

```diff
@@ -61,7 +61,7 @@ Box::Box(real l)
     box_(YY, YY) = l;
     box_(ZZ, ZZ) = l;
 
-    gmx::fillLegacyMatrix(box_, legacyMatrix_);
+    gmx::fillLegacyMatrix(box_, oldMatrix_);
```

Comment:
The name oldMatrix seems overly normative and anyway I am curious why you are undertaking these changes that are not asked for while not addressing review comments.

Reply:
Oops! That was an intermediate change when I was trying to resolve some name conflicts. Corrected.

And I am addressing the review comments. Please take a look at `NbvSetupUtils`. All the functions have been explicitly separated now. If we can merge this one, then the builder PR will begin review as that involves renaming.